src/_services/user.service.ts

```
@@ -1,17 +1,29 @@
 import {Injectable} from '@angular/core';
 
-import {User} from '../_models';
-import {HttpClient} from '@angular/common/http';
+import {User} from '../_models/User';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
 import {Observable} from 'rxjs/Observable';
 import {environment} from '../environments/environment';
+import {catchError} from 'rxjs/operators';
+import {of, throwError} from 'rxjs';
 
-@Injectable()
+@Injectable({providedIn: 'root'})
 export class UserService {
   constructor(private http: HttpClient) {
   }
 
-  create(user: User): Observable<User> {
+  create(user: User): Observable<User | null> {
     const url = environment.backendUrl + 'user/register';
-    return this.http.post<User>(url, user);
+
+    return this.http.post<User | null>(url, user).pipe(
+      catchError((error: HttpErrorResponse) => {
+        console.error(error);
+        if (error.status === 400) {
```

Probably we always should throw the error.